### PR TITLE
Make tabs resolve in different routes

### PR DESCRIPTION
**Pull request by dimitur2204**
## Motivation and context

The change is required so that the url matches where you are on the profile page.

#695 

## Screenshots:

| Before              | After              |
| ------------------- | ------------------ |
| |tabs-resolve|

## Testing

**New environment variables**:

- `env var` : env var details

**New dependencies**:

- `dependency` : dependency details

**New dev dependencies**:

- `dependency` : dependency details

### CODE REVIEW

File: src/pages/profile/index.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+import ProfilePage from 'components/auth/profile/ProfilePage'
+
+export const getServerSideProps: GetServerSideProps = async () => {
+  return {
+    redirect: {
+      destination: '/profile/donations',
+      permanent: false,
+    },
+  }
+}
```
Comment (kachar) [inline src/pages/profile/index.ts:11]: You can achieve that by using `next.config.js` rewrites 

Read more at https://nextjs.org/docs/api-reference/next.config.js/rewrites

Comment (kachar) [reply src/pages/profile/index.ts:11]: We also kind of miss the secured props here, no?

File: src/components/auth/profile/ProfilePage.tsx [hunk 2]
```diff
@@ -57,18 +89,27 @@ export default function ProfilePage() {
           }}>
           <h1 className={classes.h1}>Дарителски профил</h1>
           <Box sx={{ borderBottom: 1, borderColor: 'divider' }}>
-            <Tabs value={value} onChange={handleChange} aria-label="basic tabs example">
-              <Tab label="Дарения" />
-              <Tab label="Лична информация" />
-              <Tab label="Сертификати" />
-              <Tab label="Договор дарение" />
+            <Tabs value={TabToRender?.index} aria-label="basic tabs example">
+              <Link href="/profile/donations">
```
[the remark this answers is not in the document]
Comment (dimitur2204) [reply src/components/auth/profile/ProfilePage.tsx]: Done in https://github.com/podkrepi-bg/frontend/pull/703/commits/fb6522f27f263b17ebc1af24cceadf7dc4af0b66

File: src/pages/profile/index.ts (hunk 1 above)
Comment (dimitur2204) [reply src/pages/profile/index.ts]: Done in https://github.com/podkrepi-bg/frontend/pull/703/commits/fb6522f27f263b17ebc1af24cceadf7dc4af0b66

File: src/components/admin/ConfirmDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+import { Dialog, Card, CardContent, Box, Button, Typography } from '@mui/material'
+import { useTranslation } from 'next-i18next'
+
+import { ModalStore } from 'stores/dashboard/ModalStore'
+
+type Props = {
+  deleteHandler: () => void
+}
+
+export default observer(function ConfirmModal({ deleteHandler }: Props) {
```
Comment (kachar) [inline src/components/admin/ConfirmDialog.tsx:12]: This file seems to be irrelevant to this PR

File: src/pages/profile/[slug].ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import { GetServerSideProps } from 'next'
+import { securedProps } from 'middleware/auth/keycloak'
+import ProfilePage from 'components/auth/profile/ProfilePage'
+import { serverSideTranslations } from 'next-i18next/serverSideTranslations'
+
+export const getServerSideProps: GetServerSideProps = async (ctx) => {
+  const { slug } = ctx.query
+  return {
+    props: {
+      slug,
```
Comment (kachar) [inline src/pages/profile/[slug].ts:10]: We send the `slug` prop but we never use it in the `ProfilePage` component

There we use it again from `router.query.slug`

Let's leave only one of these

File: src/common/util/useCurrentPerson.ts [hunk 5]
```diff
@@ -14,7 +14,7 @@ type CurrentPerson = {
 export function useCurrentPerson() {
   const { keycloak } = useKeycloak<KeycloakInstance>()
   return useQuery<CurrentPerson>(
-    endpoints.account.me.url,
+    endpoints.account?.me?.url,
```
Comment (dimitur2204) [inline src/common/util/useCurrentPerson.ts:17]: My whole profile page was blowing up without this change here? Any idea what it might be and is it okay to leave it like that?

File: src/components/admin/ConfirmDialog.tsx (hunk 3 above)
Comment (dimitur2204) [reply src/components/admin/ConfirmDialog.tsx]: Removed in https://github.com/podkrepi-bg/frontend/pull/703/commits/e6f154268d2e9a941f0bf9bd92a2bf51f475995f

File: src/pages/profile/[slug].ts (hunk 4 above)
Comment (dimitur2204) [reply src/pages/profile/[slug].ts]: Removed passing of slug prop from the server side props in https://github.com/podkrepi-bg/frontend/pull/703/commits/e6c92cdf5b61ceeaa038c5172e7ea6c0165455df

File: src/common/util/useCurrentPerson.ts (hunk 5 above)
Comment (dimitur2204) [reply src/common/util/useCurrentPerson.ts]: Seems to be working fine now though 🤨, so I removed the conditionals. If I see this error again I will mention it.

**Comment by kachar** [comment]
@dimitur2204 

Added some updates to your branch

- restored tabs to be with default mui look'n'feel instead of bluish link like buttons
- improved the tab components so hey embed the tab title and tab slug
- extracted tabs as enum and reused them
- updates the styles of the page a bit so we use `theme` and default spacings instead of custom `px`

**Review by kachar — APPROVED** [review]
Thanks for the contribution @dimitur2204 :money_with_wings: :pray:
